presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/HivePartitionName.java

```diff
@@ -25,51 +25,42 @@
 
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static io.prestosql.plugin.hive.metastore.HiveTableName.hiveTableName;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.makePartitionName;
 import static io.prestosql.plugin.hive.util.HiveUtil.toPartitionValues;
 import static java.util.Objects.requireNonNull;
+import static org.apache.hadoop.hive.common.FileUtils.makePartName;
 
 @Immutable
 public class HivePartitionName
 {
     private final HiveTableName hiveTableName;
-    private final Optional<Table> table; // does not participate in hashCode/equals
     private final List<String> partitionValues;
-    private final Optional<String> partitionName; // does not participate in hashCode/equals
+    private final Optional<Partition> partition; // does not participate in hashCode/equals
```

Comment:
- isn't `Partition` a much bigger object? Since this is part of the cache, object size matters.
- the class name `HivePartitionName` is no longer appropriate.

Reply:
> isn't Partition a much bigger object? Since this is part of the cache, object size matters.

`Partition` should be cached in `partitionCache`. Previously we also fetched partitions when obtaining stats, but we didn't cache `Partition` objects. However, we should query only partitions that will participate in query, so they would be fetched anyway.

> the class name HivePartitionName is no longer appropriate.

what name do you suggest?

Reply:
> what name do you suggest?

`HivePartition`?